lib/widgets/topic_list.dart

```diff
@@ -0,0 +1,301 @@
+        child: Row(
+          mainAxisSize: MainAxisSize.min,
+          crossAxisAlignment: CrossAxisAlignment.center,
+          children: [
+            Icon(size: 16, iconData),
+            const SizedBox(width: 4),
+            Flexible(child: Row(
+              children: [
+                Text(
+                  stream?.name ?? zulipLocalizations.unknownChannelName,
+                  style: recipientHeaderTextStyle(context),
+                ),
+                const SizedBox(width: 10),
+                Icon(ZulipIcons.chevron_down, size: 10, color: designVariables.icon),
+              ],
+            )),
+          ]))
+    );
+  }
+}
+
+class TopicListPageBody extends StatefulWidget {
+  const TopicListPageBody({super.key, required this.streamId});
+
+  final int streamId;
+
+  @override
+  State<TopicListPageBody> createState() => _TopicListPageBodyState();
+}
+
+class _TopicListPageBodyState extends State<TopicListPageBody>
+    with PerAccountStoreAwareStateMixin<TopicListPageBody> {
+  TopicListView? _model;
+  Unreads? _unreadsModel;
+
+  @override
+  void onNewStore() {
+    _unreadsModel?.removeListener(_modelChanged);
+    final store = PerAccountStoreWidget.of(context);
+    _unreadsModel = store.unreads..addListener(_modelChanged);
+
+    _model?.dispose();
+    _model = TopicListView(store: store, streamId: widget.streamId);
+    _model!.addListener(_modelChanged);
+    _fetchTopics();
+  }
+
+  Future<void> _fetchTopics() async {
+    try {
+      await _model!.fetchTopics();
+      if (_model!.hasError && mounted) {
+        final zulipLocalizations = ZulipLocalizations.of(context);
+        showErrorDialog(
+          context: context,
+          title: zulipLocalizations.errorFetchingTopics,
+          message: _model!.errorMessage);
+      }
+    } catch (e) {
+      if (!mounted) return;
+      final zulipLocalizations = ZulipLocalizations.of(context);
+      showErrorDialog(
+        context: context,
+        title: zulipLocalizations.errorFetchingTopics,
+        message: e.toString());
+    }
+  }
+
+  void _modelChanged() {
+    setState(() {
+      // The actual state lives in the models.
+      // This method was called because they just changed.
+    });
+  }
+
+  @override
+  void dispose() {
+    _unreadsModel?.removeListener(_modelChanged);
+    _model?.dispose();
+    super.dispose();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    if (_model == null || _model!.isLoading) {
+      return const Center(child: CircularProgressIndicator());
+    }
+
+    if (_model!.topics == null || _model!.topics!.isEmpty) {
+      return Center(
+        child: Text(ZulipLocalizations.of(context).noTopicsInChannel,
+          style: TextStyle(
+            color: DesignVariables.of(context).labelMenuButton,
+            fontSize: 16,
+          )),
+      );
+    }
+
+    final sortedTopics = List<GetStreamTopicsEntry>.from(_model!.topics!);
+    sortedTopics.sort((a, b) => b.maxId.compareTo(a.maxId));
```

Comment:
It probably makes more sense to sort the topics with the model. Because if the topics didn't change since last build, re-sorting them is not necessary.